Automattic/wp-calypso
# Add/checklist toggle affordance

~**Note: This PR is still a work in progress. I have the visual styles applied but am waiting on another PR to merge to finalize the functionality. For more details checkout https://github.com/Automattic/zelda-private/issues/51**~

#### Changes proposed in this Pull Request

This PR:
- Adds an affordance to the checklist tasks so people know they can toggle them
- Synchronizes the Jetpack and WP.com checklists to both have the expand/collapse functionality

###### Screens
![image](https://user-images.githubusercontent.com/6981253/58983386-adb1ae80-87a4-11e9-9608-38ee1d576163.png)


![image](https://user-images.githubusercontent.com/6981253/58983386-adb1ae80-87a4-11e9-9608-38ee1d576163.png)

#### Testing instructions

In your browser console: `localStorage.setItem( 'debug', 'calypso:analytics:tracks' )`, and reload.

WordPress.com
* Create a new site
* Visit `/checklist/{your-url}` (First task should be open while others are closed. )
* Click on the task title to toggle the view.
* In the browser console, verify that expanding a task fires a `calypso_checklist_task_expand` Tracks event, with the correct `step_name` prop set.

Jetpack
* Create a new site with Jurrasic ninja
* Visit `/plans/my-plan/{your-url}` (First task should be open while others are closed. )
* Click on the task title to toggle the view.
* In the browser console, verify that expanding a task fires a `calypso_checklist_task_expand` Tracks event, with the correct `step_name` prop set.

WooCommerce
* Logged-in, navigate to http://calypso.localhost:3000/start/site-type
* Select 'Online Store' box and click continue.
* Go thru the usual NUX flow (select a domain, pick the eCommerce plan, and pay for your plan. You can skip adding G Suite).
* When presented with the purple 'Thank You' page, click the 'Create your Store' button.
* (The 'Building your Store' step can take a while.)
* Click 'Set up my Store'. Another 'Building your Store' step follows.
* When setting your address, it's important to enter a US or Canadian one.
* You're now presented with the Store checklist! From now on, you can always return to this site by navigating to `http://calypso.localhost:3000/store/<site>`
* Verify that the checklist works -- try toggling individual tasks

Fixes https://github.com/Automattic/zelda-private/issues/51

#### Follow-up Work

This is really looking a lot like a collection of [`FoldableCard`](http://calypso.localhost:3000/devdocs/design/foldable-card) components now, so it might make sense to actually use those to implement the checklist (their props interface permitting)
